pandas/core/missing.py

```diff
@@ -21,11 +21,16 @@ def mask_missing(arr, values_to_mask):
     Return a masking array of same size/shape as arr
     with entries equaling any member of values_to_mask set to True
     """
-    if not isinstance(values_to_mask, (list, np.ndarray)):
+    if isinstance(values_to_mask, np.ndarray):
+        mask_type = values_to_mask.dtype.type
+    elif isinstance(values_to_mask, list):
```

Comment:
you can change this entire test to:
```
# import at top if its not
from pandas._libs.lib import infer_dtype
....
inferred = infer_dtype(values_to_mask)
if inferred in ['string', 'unicode']:
    mask_type = np.object
else:
    mask_type = np.asarray(values_to_mask).dtype
```

I *think* this will work.